bert/model.py

```diff
@@ -0,0 +1,246 @@
+            TransformerEncoderLayer(dims, num_heads, mlp_dims)
+            for i in range(num_layers)
+        ]
+
+    def __call__(self, x, mask):
+        for l in self.layers:
+            x = l(x, mask)
+
+        return x
+
+
+class BertEmbeddings(nn.Module):
+    def __init__(self, config: ModelArgs):
+        self.word_embeddings = nn.Embedding(config.vocab_size, config.intermediate_size)
+        self.token_type_embeddings = nn.Embedding(2, config.intermediate_size)
+        self.position_embeddings = nn.Embedding(
+            config.max_position_embeddings, config.intermediate_size
+        )
+        self.norm = nn.LayerNorm(config.intermediate_size, eps=config.layer_norm_eps)
+
+    def __call__(self, input_ids: mx.array, token_type_ids: mx.array) -> mx.array:
+        words = self.word_embeddings(input_ids)
+        position = self.position_embeddings(
+            mx.broadcast_to(mx.arange(input_ids.shape[1]), input_ids.shape)
+        )
+        token_types = self.token_type_embeddings(token_type_ids)
+
+        embeddings = position + words + token_types
+        return self.norm(embeddings)
+
+
+class Bert(nn.Module):
+    def __init__(self, config: ModelArgs):
+        self.embeddings = BertEmbeddings(config)
+        self.encoder = TransformerEncoder(
+            num_layers=config.num_hidden_layers,
+            dims=config.intermediate_size,
+            num_heads=config.num_attention_heads,
+        )
+        self.pooler = nn.Linear(config.intermediate_size, config.vocab_size)
+
+    def __call__(
+        self,
+        input_ids: mx.array,
+        token_type_ids: mx.array,
+        attention_mask: mx.array | None = None,
+    ) -> tuple[mx.array, mx.array]:
+        x = self.embeddings(input_ids, token_type_ids)
+        y = self.encoder(x, attention_mask)
+        return y, mx.tanh(self.pooler(y[:, 0]))
+
+
+def run(bert_model: str, mlx_model: str):
+    batch = [
+        "This is an example of BERT working on MLX.",
+        "A second string",
+        "This is another string.",
+    ]
+
+    model = Bert(model_configs[bert_model])
+
+    weights = mx.load(mlx_model)
+    weights = tree_unflatten(list(weights.items()))
+    weights = tree_map(lambda p: mx.array(p), weights)
+
+    model.update(weights)
+
+    tokenizer = AutoTokenizer.from_pretrained(bert_model)
+
+    tokens = tokenizer(batch, return_tensors="np", padding=True)
+    tokens = {key: mx.array(v) for key, v in tokens.items()}
+
+    mlx_output, mlx_pooled = model(**tokens)
+    mlx_output = numpy.array(mlx_output)
+    mlx_pooled = numpy.array(mlx_pooled)
+
+    print("MLX BERT:")
+    print(mlx_output)
+
+    print("\n\nMLX Pooled:")
+    print(mlx_pooled[0, :20])
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Convert BERT weights to MLX.")
+    parser.add_argument(
+        "--bert-model",
+        type=str,
+        default="bert-base-uncased",
+        help="The huggingface name of the BERT model to save.",
```

Comment:
I like the `choices` kwarg for this sort of thing. You could add it and list the available models and it will handle making sure the user supplies one in the list.